src/NotificationTargetCommonITILObject.php

```diff
@@ -1678,6 +1701,96 @@ public function getDataForObject(CommonDBTM $item, array $options, $simple = fal
         return $data;
     }
 
+    protected function getActorData(CommonDBTM $actor, int $actortype, string $key_prefix): array
+    {
+        $data = [
+            sprintf('##%s.itemtype', $key_prefix)  => get_class($actor),
+            sprintf('##%s.actortype', $key_prefix) => $actortype,
+            sprintf('##%s.id', $key_prefix)        => $actor->getID(),
+            sprintf('##%s.name', $key_prefix)      => '',
+            sprintf('##%s.location', $key_prefix)  => '',
+            sprintf('##%s.title', $key_prefix)     => '',
+            sprintf('##%s.category', $key_prefix)  => '',
+            sprintf('##%s.email', $key_prefix)     => '',
+            sprintf('##%s.mobile', $key_prefix)    => '',
+            sprintf('##%s.phone', $key_prefix)     => '',
+            sprintf('##%s.phone2', $key_prefix)    => '',
+            sprintf('##%s.fax', $key_prefix)       => '',
+            sprintf('##%s.website', $key_prefix)   => '',
+            sprintf('##%s.address', $key_prefix)   => '',
+            sprintf('##%s.postcode', $key_prefix)  => '',
+            sprintf('##%s.town', $key_prefix)      => '',
+            sprintf('##%s.state', $key_prefix)     => '',
+            sprintf('##%s.country', $key_prefix)   => '',
+            sprintf('##%s.comments', $key_prefix)  => '',
+            sprintf('##%s.type', $key_prefix)      => '',
+        ];
+        switch (get_class($actor)) {
+            case User::class:
+                $data[sprintf('##%s.name', $key_prefix)] = $actor->getName();
+                
+                if ($actor->getField('locations_id')) {
+                    $data[sprintf('##%s.location', $key_prefix)]
+                                = Dropdown::getDropdownName(
+                                    'glpi_locations',
+                                    $actor->getField('locations_id')
+                                );
+                    $location = new Location();
+                    if ($location->getFromDB($actor->getField('locations_id'))) {
+                        $data[sprintf('##%s.address', $key_prefix)]  = $location->getField('address');
+                        $data[sprintf('##%s.postcode', $key_prefix)] = $location->getField('postcode');
+                        $data[sprintf('##%s.town', $key_prefix)]     = $location->getField('town');
+                    }
+                }
+
+                if ($actor->getField('usertitles_id')) {
+                    $data[sprintf('##%s.title', $key_prefix)]
```

Comment:
To prevent any ambiguity, the key should correspond to the full dropdown class name.
```suggestion
                    $data[sprintf('##%s.usertitle', $key_prefix)]
```